src/libertem/web/events.py

```diff
@@ -64,3 +62,5 @@ async def open(self):
 
     def on_close(self):
         self.event_registry.remove_handler(self)
+        if self.event_registry.get_number_of_handlers() < 1: # Still isnt a perfect solution - refreshing a page also means close I guess
+            self.state.deregister()
```

Comment:
I added this so that whenever a user closes all connections, we get rid of the session state. But one issue I faced is that if there is only one tab open and the user refreshes it, then it will also delete the session and all changes made to the dataset. One solution I can think of is having a timer for a couple of seconds before the actual delete happens. So when the browser tab is refreshed, the reconnect should cancel the delete timer. What do you think?